pysat/instruments/methods/testing.py

```diff
@@ -515,3 +518,84 @@ def define_range():
                  'angle': [0.0, 2.0 * np.pi]}
 
     return def_range
+
+
+def create_files(inst, start, stop, freq=None, use_doy=True, root_fname=None,
+                 version=False, content=None, timeout=None):
+    """Create a file set using the year and day of year.
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        A test instrument, used to generate file path
+    start : dt.datetime
+        The date for the first file to create
+    stop : dt.datetime
+        The date for the last file to create
+    freq : str
+        Frequency of file output.  Ex: '1D', '100min'
+        (default=None)
+    use_doy : bool
+        If True, use Day of Year (doy)
+        If False, use month / day
+        (default=True)
+    root_fname : str
+        The format of the file name to create.  Uses standard pysat variables.
+        Ex: 'pysat_testing_junk_{year:04d}_{day:03d}.txt'
+        (default=None)
+    version : bool
+        If True, iterate over version / revision / cycle
+        If False, ignore version / revision / cycle
+        (default=False)
+    content : str
+        Custom text to write to temporary files
+        (default=None)
+    timeout : float
+        Time is seconds to lock the files being created.  If None, no timeout is
+        used.  (default=None)
+
```

Comment:
Could you add an example here or in the docs?

Reply:
Added an example. One of the string inputs here defaults to `None`, and if `None` is replaced with a real default in the method. However, that practical default string is  long, `''.join(('pysat_testing_junk_{year:04d}_gold_{day:03d}_',
                              'stuff.pysat_testing_file'))`

How should I write up the default?

Reply:
I would replace your second example with the default that uses DOY, and have the first one with the custom name have `use_doy`=False.  That way there's only two examples and one of them uses all the possible defaults.

Out of curiosity, why such a long default name?  Seems unnecessarily awkward.  Also not really best practice to put a Constellation name in the general default.

Reply:
Maybe 'pysat_testing_{year:04d}_{day:03d}.txt' is a better default name?  Or have a default that includes all the possible file specification options (e.g., cycle, version, etc.)?

Reply:
> 
> Out of curiosity, why such a long default name? Seems unnecessarily awkward. Also not really best practice to put a Constellation name in the general default.

That default was set early in pysat's history. Creating files for testing also means deleting files. I wanted to make sure I didn't erase something on a user's computer I wasn't supposed to. Happy to see the intended awkward and long filenames come across as such. The default value predates Gold the Constellation by a healthy amount.

I've gone through the code and I didn't find anything in pysat itself that uses `create_files` using the default string name (anymore). So, yeah, looks like the default can be updated.

Reply:
The default has been updated. I did not include `version` etc since by default `create_files` doesn't make files with that info. Consistency! 😉